################################################################################
SHORT DESCRIPTION: 
################################################################################
Remove creation of unused log files.

################################################################################
LONG DESCRIPTION: 
################################################################################
Remove creation of unused log files.

################################################################################
CHANGELOG 
################################################################################
Author(s):
Mahantesh Salimath <salimathm@ainfosec.com>

################################################################################
REMOVAL 
################################################################################

################################################################################
UPSTREAM PLAN
################################################################################
Not sure, probably not.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################

################################################################################
PATCHES 
################################################################################
--- a/drivers/tapdisk-log.c
+++ b/drivers/tapdisk-log.c
@@ -73,74 +73,6 @@ tlog_logfile_vprint(const char *fmt, va_
 	tapdisk_logfile_vprintf(&tapdisk_log.logfile, fmt, ap);
 }
 
-static void __printf(1, 2)
-tlog_logfile_print(const char *fmt, ...)
-{
-	va_list ap;
-
-	va_start(ap, fmt);
-	tlog_logfile_vprint(fmt, ap);
-	va_end(ap);
-}
-
-#define tlog_info(_fmt, _args ...)					\
-	tlog_logfile_print("%s: "_fmt, tapdisk_log.ident, ##_args)
-
-/**
- * Closes the log file.
- *
- * @param keep if set to true the log file is never removed. NB if an error has
- * occurred or a USR1 has been received the log is always kept, independently
- * of whether @keep is set to false.
- */
-static void
-tlog_logfile_close(bool keep)
-{
-	td_logfile_t *logfile = &tapdisk_log.logfile;
-
-	if (tapdisk_log.precious || tapdisk_log.errors)
-		keep = true;
-
-	tlog_info("closing log, %lu errors", tapdisk_log.errors);
-
-	tapdisk_logfile_close(logfile);
-
-	if (!keep)
-		tapdisk_logfile_unlink(logfile);
-}
-
-static int
-tlog_logfile_open(const char *name, int level)
-{
-	td_logfile_t *logfile = &tapdisk_log.logfile;
-	int mode, err;
-
-	err = mkdir(TLOG_DIR, 0755);
-	if (err) {
-		err = -errno;
-		if (err != -EEXIST)
-			goto fail;
-	}
-
-	err = tapdisk_logfile_open(logfile, TLOG_DIR, name, TLOG_LOGFILE_BUFSZ);
-	if (err)
-		goto fail;
-
-	mode = (level == TLOG_DBG) ? _IOLBF : _IOFBF;
-
-	err = tapdisk_logfile_setvbuf(logfile, mode);
-	if (err)
-		goto fail;
-
-	tlog_info("log start, level %d", level);
-
-	return 0;
-
-fail:
-	tlog_logfile_close(false);
-	return err;
-}
-
 static void
 tlog_syslog_close(void)
 {
@@ -196,15 +128,11 @@ tlog_open(const char *name, int facility
 	tapdisk_log.ident = tapdisk_syslog_ident(name);
 	tapdisk_log.facility = facility;
 
-	if (!tapdisk_log.name || !tapdisk_log.ident) {
+	if (!tapdisk_log.ident) {
 		err = -errno;
 		goto fail;
 	}
 
-	err = tlog_logfile_open(tapdisk_log.name, level);
-	if (err)
-		goto fail;
-
 	err = tlog_syslog_open(tapdisk_log.ident, facility);
 	if (err)
 		goto fail;
@@ -220,13 +148,6 @@ fail:
 int
 tlog_reopen(void)
 {
-	int err;
-
-	tlog_logfile_close(true);
-	err = tlog_logfile_open(tapdisk_log.name, tapdisk_log.level);
-	if (err)
-		return err;
-
 	tlog_syslog_close();
 	return tlog_syslog_open(tapdisk_log.ident, tapdisk_log.facility);
 }
@@ -237,7 +158,6 @@ tlog_close(void)
 	DPRINTF("tapdisk-log: closing after %lu errors\n",
 		tapdisk_log.errors);
 
-	tlog_logfile_close(false);
 	tlog_syslog_close();
 
 	free(tapdisk_log.ident);
